backend/src/main/kotlin/com/tngtech/apicenter/backend/connector/database/service/SpecificationDatabaseService.kt

```diff
@@ -24,26 +24,29 @@ class SpecificationDatabaseService constructor(
     override fun save(specification: Specification) {
         val specificationEntity = specificationEntityMapper.fromDomain(specification)
 
-        if (specificationRepository.existsByTitle(specificationEntity.title)) {
-            val existingEntity = specificationRepository.findByTitle(specificationEntity.title)
-
+        val entityToStore = specificationRepository.findById(specificationEntity.id).map {
+            existingEntity ->
```

Comment:
I find this a bit long for a lambda. Either extract it into a nice method or restructure it to not require a lambda (or at least such a long one).

By the way, instead of naming the parameter, you can also just use `it`.